packages/web/src/components/inbox-settings-modal/InboxSettingsModalNew.tsx

```diff
@@ -99,62 +89,90 @@ export const InboxSettingsModalNew = () => {
-  }
+  const initialValues = useMemo(() => {
+    return transformPermitListToMap(
+      permissions?.permit_list ?? [ChatPermission.ALL]
+    )
+  }, [permissions])
 
   return (
-    <Formik<InboxSettingsFormValues>
-      initialValues={initialValues}
-      onSubmit={handleSave}
-    >
-      {({ submitForm }) => (
-        <Modal onClose={handleClose} isOpen={isVisible} size='medium'>
-          <ModalHeader onClose={handleClose}>
-            <ModalTitle title={messages.title} icon={<IconMessage />} />
-          </ModalHeader>
-          <ModalContent>
-            <InboxSettingsModalFields />
-          </ModalContent>
-          <ModalFooter>
-            <Flex gap='xl' flex={1}>
-              <Button variant='secondary' fullWidth>
-                {messages.cancel}
-              </Button>
-              <Button
-                variant='primary'
-                isLoading={permissionStatus === Status.LOADING && showSpinner}
-                type='submit'
-                onClick={submitForm}
-                fullWidth
-              >
-                {messages.save}
-              </Button>
-            </Flex>
-          </ModalFooter>
-        </Modal>
+    <Modal onClose={handleClose} isOpen={isVisible} size='medium'>
+      <ModalHeader onClose={handleClose}>
+        <ModalTitle title={messages.title} icon={<IconMessage />} />
+      </ModalHeader>
+      {statusIsNotFinalized(permissionsStatus) ? (
+        <Flex
+          justifyContent='center'
+          alignItems='center'
+          m='xl'
+          p='4xl'
+          h='unit10'
+          css={{
+            opacity: 0.5
+          }}
+        >
+          <LoadingSpinner />
+        </Flex>
+      ) : (
+        <Formik<InboxSettingsFormValues>
+          initialValues={initialValues}
+          onSubmit={handleSave}
+        >
+          {({ submitForm }) => (
+            <>
+              <ModalContent>
+                <InboxSettingsModalFields
+                  permissionsStatus={permissionsStatus}
+                />
+              </ModalContent>
+              <ModalFooter>
+                <Flex gap='xl' flex={1}>
+                  <Button variant='secondary' fullWidth onClick={handleClose}>
+                    {messages.cancel}
+                  </Button>
+                  <Button
+                    variant='primary'
+                    isLoading={
+                      savePermissionStatus === Status.LOADING && showSpinner
+                    }
+                    type='submit'
+                    onClick={submitForm}
+                    fullWidth
+                  >
+                    {messages.save}
+                  </Button>
+                </Flex>
+              </ModalFooter>
+            </>
+          )}
+        </Formik>
       )}
-    </Formik>
+    </Modal>
   )
 }
 
-const InboxSettingsModalFields = () => {
-  const [allowAllField] = useField({ name: 'allowAll', type: 'checkbox' })
+const InboxSettingsModalFields = ({
```

Comment:
this file might be big enough now that we should just move these out. Thoughts?

Reply:
ehhh i'm a fan of co-location personally